### ISSUE TITLE
Removing `StateKeysMaxChunks()` from the `chain.Action` interface

### ISSUE THREAD

**Issue by RodrigoVillar** [ava-labs/hypersdk#1599]
A common piece of DevX feedback is regarding the `StateKeysMaxChunks()` method that all actions must implement. In particular, developers find `StateKeysMaxChunks()` to be both unintuitive and redundant. Therefore, I propose that we remove `StateKeysMaxChunks()` from the `chain.Action` interface. 

Doing this is relatively straightforward except for the case of `EstimateUnits()`, which is the only function dependent on `StateKeysMaxChunks()`.  To fix this issue, we would implement the following pseudocode:

```
stateKeysMaxChunks := make([]uint16, 0)
for _, action := range actions {
    chunks := make([]uint16, 0)
    stateKeys := action.StateKeys(actor)
    
    for k := range stateKeys {
        chunk := GetChunkSize(k)
        chunks := append(chunks, chunk)
    }
    stateKeysMaxChunks := append(stateKeysMaxChunks, chunks)
}
```

`GetChunkSize` is a helper function which extracts the last two bytes from the state key.

This would then leave us with figuring out how to derive `actor` in `EstimateUnits`. After discussing with @aaronbuchwald , we can derive the actor by the following:

- Signing an arbitrary message with the provided `authFactory` argument
- Since the signed message is of type `Auth`, we can call the `Actor()` method on it

## tl;dr

- We would remove `StateKeysMaxChunks()` from the `chain.Action` interface 
- We remove `StateKeysMaxChunks()` from both MorpheusVM and VMWithContracts
- We would modify `StateKeys()` by removing the action ID parameter
- We would modify `EstimateUnits()` in `chain/transaction.go` with the pseudocode above

**Comment by aaronbuchwald** [comment]
What do you think of including `GetChunkSize` on the `state.Keys` type itself?

Something like:

```golang
func (k Keys) ChunkSizes() []uint16 { ... }
```

**Comment by aaronbuchwald** [comment]
Also, would it make more sense to add `Address() codec.Address` to the `AuthFactory` interface, so that we don't need the throwaway step of generating an `Auth` type and extracting the address from it?
